File: pypp_cli/test_dir/cpp/src/dicts/looping.cpp
#include "dicts/looping.h"
#include "py_dict.h"
#include "py_enumerate.h"
#include "py_list.h"
#include "py_str.h"
#include "py_zip.h"
#include "pypp_util/print.h"

namespace me {
void dict_looping_fn() {
    pypp::print(pypp::PyStr("DICT LOOPING RESULTS:"));
    pypp::PyDict<int, int> d = {{0, 1}, {1, 2}, {2, 3}};
    for (const auto &[i, k] : pypp::PyEnumerate(d.keys())) {
        pypp::print(pypp::PyStr(std::format("{}: {} -> {}", i, k, d[k])));
    }
    for (const auto &[i, k, v] :
         pypp::PyZip(pypp::PyList({1, 2, 3}), d.keys(), d.values())) {
        pypp::print(pypp::PyStr(std::format("{}: {} -> {}", i, k, v)));
    }
}

} // namespace me

File: pypp_cli/test_dir/src/dicts/looping.py
def dict_looping_fn():
    print("DICT LOOPING RESULTS:")
    # over a dict's items
    d: dict[int, int] = {0: 1, 1: 2, 2: 3}
    for i, k in enumerate(d.keys()):
        print(f"{i}: {k} -> {d[k]}")
    for i, k, v in zip([1, 2, 3], d.keys(), d.values()):
        print(f"{i}: {k} -> {v}")
